subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.groovy

```diff
@@ -215,16 +217,22 @@ ${globalExclusions(this.effectivePom)}
         def artifactId = project.artifactId.text()
         buffer.insert(0, ":${artifactId}")
         //we don't need the top-level parent in gradle, so we stop on it
-        if (project.parent.artifactId.text() != allProjects[0].artifactId.text()) {
+        if (getModuleIdentifier(project.parent) != getModuleIdentifier(allProjects[0])) {
             def parentInBuild = allProjects.find { proj ->
-                proj.artifactId.text() == project.parent.artifactId.text()
+                getModuleIdentifier(proj) == getModuleIdentifier(project.parent)
             }
             if (parentInBuild) {
                 generateFqn(parentInBuild, allProjects, buffer)
             }
         }
     }
 
+    private ModuleIdentifier getModuleIdentifier(def project) {
```

Comment:
:o: I'd type `project` here for better understanding. In the long run we might want to convert the class into Java.

Reply:
Good catch. I felt inconvenient as well when I first saw this code.